sapi_demo/src/Plugin/Statistics/ActionHandler/EntityInteractionTracker.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\sapi_demo\Plugin\Statistics\ActionHandler;
+
+use Drupal\sapi\ActionHandlerInterface;
+use Drupal\sapi\ActionHandlerBase;
+use Drupal\sapi\ActionTypeInterface;
+use Drupal\sapi\Plugin\Statistics\ActionType\EntityInteraction;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+
+/**
+ * This is a SAPI handler plugin used to track any entity interactions (view, update and create).
+ *
+ * @ActionHandler(
+ *  id = "entity_interaction_tracker",
+ *  label = "Track any entity interactions"
+ * )
+ */
+class EntityInteractionTracker extends ActionHandlerBase implements ActionHandlerInterface{
+
+  /**
+   * EntityTypeManager used to get entity storage for sapi_data items, which is
+   * used to create and edit sapi_data items from tracking.
+   *
+   * @protected Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * EntityInteractionTracker constructor.
+   *
+   * @param array $configuration
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(ActionTypeInterface $action){
+
+    /**
+     * Only acts if $action is an EntityInteraction plugin type
+     */
+    if (!($action instanceof EntityInteraction)) {
+      return;
+    }
+
+    /**
+     * @var string $entity_type
+     *   Label of Entity
+     */
+    $entity_type = $action->getEntity()->getEntityTypeId();
+
+    /**
+     * @var string $entity_id
+     *   ID of Entity
+     */
+    $entity_id = $action->getEntity()->id();
+
+    /**
+     * @var string $interaction_type
+     *   Interaction type to Entity(View, Update, Create)
+     */
+    $interaction_type = $action->getAction();
+
+    /**
+     * @var string $user_id
+     *   UID of user who interacted to Entity
+     */
+    $user_id = $action->getAccount()->id();
+
+    /**
+     * @var EntityStorageInterface $sapiDataStorage
+     */
```

Comment:
No change required:  if your @var includes a "description of what it does" then make it multiple lines, otherwise it can be a single line  /** @var type $variable */  statement.